src/plugins/core/conditional_format.ts

```diff
@@ -58,37 +58,99 @@ export class ConditionalFormatPlugin
 
   readonly cfRules: { [sheet: string]: ConditionalFormatInternal[] } = {};
 
-  loopThroughRangesOfSheet(sheetId: UID, applyChange: ApplyRangeChange) {
-    for (const rule of this.cfRules[sheetId]) {
-      if (rule.rule.type === "DataBarRule" && rule.rule.rangeValues) {
-        const change = applyChange(rule.rule.rangeValues);
-        switch (change.changeType) {
-          case "REMOVE":
-            this.history.update(
-              "cfRules",
-              sheetId,
-              this.cfRules[sheetId].indexOf(rule),
-              "rule",
-              //@ts-expect-error
-              "rangeValues",
-              undefined
-            );
-            break;
-          case "RESIZE":
-          case "MOVE":
-          case "CHANGE":
+  loopThroughAllRanges(applyChange: ApplyRangeChange) {
+    for (const sheetId of Object.keys(this.cfRules)) {
```

Comment:
Equivalent but without `Object.keys` (which allocates a new array, better avoid it if we can)
```suggestion
    for (const sheetId in this.cfRules) {
```